### DISCUSSION TITLE
Raspap Couldn't Get Ip Address help a newbie please 😢

### DISCUSSION THREAD

**Discussion by astrawnuts** [RaspAP/raspap-webgui#1381] [Q&A] [answered]
Hi everyone, I feel like I have come a long way with my RaspAP setup. Would really appreciate any help on this to know where i am going wrong. Im reasonably new to linux so dont have that much understanding. 

when trying to connect a device to the hotspot it says couldnt get ip address. 

My Setup:
OS: Server Ubuntu 22.04.3 LTS
Release:        22.04
Codename:       jammy

- Using TP Link Wireless USB Adapter with Intel Nuc (as far as I can see drivers have been installed).
- Intel NUC connects to ethernet cable
- USB Wifi Adapater works as hotspot/access point
- when trying to connect phones to the access point it says couldn't get ip address (especially true in android).
- I have followed the guide to the best of my understanding

```
$ lsusb

 TP-Link 


```

**Comment by billz** [comment, marked as answer]
It appears your AP interface has an automatically assigned IP from the `169.254.x.x` private address range. This is a failover mechanism for local IPv4 networks when clients are unable to obtain an IP address from a DHCP server.

The AP interface requires a static IPv4 address, eg.:

https://github.com/RaspAP/raspap-webgui/blob/470debb945e797af75b16ac234478907c694ecf9/config/dhcpcd.conf#L13-L16

This is done automatically by RaspAP's [default settings](https://docs.raspap.com/defaults/), however it doesn't know how to handle the [predictable network interface name](https://www.freedesktop.org/wiki/Software/systemd/PredictableNetworkInterfaceNames/) assigned to your `wlx[...]` USB device. 

If you'd like to use your external wireless adapter as the AP interface, be sure to assign a static IP to this interface in the **DHCP server settings** UI. Save the settings, restart the AP, confirm your static IP, then try connecting clients again.

Protip: enable DHCP request logging on the **DHCP Server > Logging** tab for additional debug info.
